guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -154,24 +163,30 @@ protected int nextEscapeIndex(CharSequence csq, int start, int end) {
    * @throws NullPointerException if {@code string} is null
    * @throws IllegalArgumentException if invalid surrogate characters are encountered
    */
-  protected final String escapeSlow(String s, int index) {
-    int end = s.length();
+  @SuppressWarnings(value = {"assignment.type.incompatible",//use of indexInternal = index cause this.
+          "argument.type.incompatible"//(203,64): escaped.length is length of escaped, so it should be already
+          //inferred as @LTEqLengthOf("escaped")
+  })
+  protected final String escapeSlow(String s, @IndexOrHigh("#1") int index) {
+    @LengthOf("s") int end = s.length();
 
     // Get a destination buffer and setup some loop variables.
     char[] dest = Platform.charBufferFromThreadLocal();
-    int destIndex = 0;
-    int unescapedChunkStart = 0;
+    @IndexOrHigh(value = {"s", "dest"}) int destIndex = 0;
+    int destIndexInternal = destIndex;
```

Comment:
Only some occurences of `destIndex` are replaced by `destIndexInternal`. This will almost certainly cause this code not to work correctly.
When changing the code, one needs to be careful not to change the behavior.

Anyway, I don't think adding these variables is what is needed here. In `codePointAt`, you had to do that because the method required the parameter to be `@IndexFor`, but changed that variable internally to `@IndexOrHigh` (less specific type). That is not the case here.